code/__HELPERS/_string_lists.dm

```diff
@@ -1,27 +1,40 @@
 #define pick_list(FILE, KEY) (pick(strings(FILE, KEY)))
+#define pick_list_replacements(FILE, KEY) (strings_replacement(FILE, KEY))
 
 var/global/list/string_cache
+//This is a hack to track which file is being processed for string subkey
+//replacements, yes it's ugly, I don't know an easier way though
+var/global/string_filename_current_key

```

Comment:
This is an ugly hack to make up for being unable to pass the filename through to the string_subkey_lookup function

if anyone knows a better way I'm open to it.